src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConditionalWeakTable.cs

```diff
@@ -538,7 +538,17 @@ internal int FindEntry(TKey key, out object? value)
             {
                 Debug.Assert(key != null); // Key already validated as non-null.
 
-                int hashCode = RuntimeHelpers.GetHashCode(key) & int.MaxValue;
+                int hashCode = RuntimeHelpers.TryGetHashCode(key);
+
+                if (hashCode == 0)
```

Comment:
Is it certain that 0 is not a valid hashcode?

Reply:
This PR includes changes to Mono to ensure that property. CoreCLR and NativeAOT already had that property.

CoreCLR:

https://github.com/AustinWise/runtime/blob/austin/TryGetHashCode/src/coreclr/vm/object.cpp#L37

NativeAOT:

https://github.com/AustinWise/runtime/blob/austin/TryGetHashCode/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/ObjectHeader.cs#L123-L130